moveit_core/robot_model/test/test.cpp

```diff
@@ -128,6 +128,50 @@ TEST(SiblingAssociateLinks, SimpleYRobot)
   }
 }
 
+TEST(FloatingJointTest, interpolation_test)
+{
+  // Create a simple floating joint model with some dummy parameters (these are not used by the test)
+  moveit::core::FloatingJointModel fjm("joint", 0, 0);
+
+  double jv1[7];
+  double jv2[7];
+  double intp[7];
+  random_numbers::RandomNumberGenerator rng;
```

Comment:
There is currently some ongoing refactoring around RNG, see #1267. @sjahr any thoughts if we should merge this and refactor later?

Reply:
It's not half-theta at all, that really is the angle between the two quaternions. The halving only shows up when you when you convert between Quaternions and axis-angle stuff to my knowledge

Reply:
100% agree that just calling into Eigen would be way easier than DIY-ing the math ourselves; I just didn't feel confident suggesting a larger rewrite than was necessary to fix the bug.

Reply:
I think we can merge this now b/c it's an obvious improvement. Thanks! especially for the new test

I'm still not sure I agree on the factor of 2 though. We can make an issue and figure it out later.

Reply:
Oh I think I misread your website. Lemme take a look at what they're doing, Quaternions are tricky

Reply:
Ok, find a source here that features the *2 as well https://nl.mathworks.com/matlabcentral/answers/415936-angle-between-2-quaternions

Reply:
Here's a *fantastic* introduction to quaternions that only *slightly* breaks my brain: https://eater.net/quaternions

Reply:
https://math.stackexchange.com/questions/4053346/the-angle-between-two-quaternion-vectors-is-equal-to-half-the-angle-between-thei

> that given two quaternions q1,q2∈R4, the angle between the 4D quaternion vectors is equal to half the angle between their corresponding 3D orientations.

Oh my god the angle is half the angle... I think this is a definitions issue. Do we want the angle between rotations, or do we want the angle between the quaternions?

On the referenced Wikipedia article: "Compute Ω as the angle [subtended](https://en.wikipedia.org/wiki/Subtended_angle) by the arc, so that cos Ω = p0 ∙ p1, the n-dimensional [dot product](https://en.wikipedia.org/wiki/Dot_product) of the unit vectors from the origin to the ends."

So (if I read this correctly) it's the angle between the vectors (quaternions are basically 4D vectors), not the angle between the rotations that the quaternions represent. Therefore, it's theta and not half-theta.

This also reinforced my belief that we really should just let Eigen handle this...

Reply:
In light of that explanation, this code here (https://github.com/ros-planning/moveit2/blob/main/moveit_core/robot_model/src/floating_joint_model.cpp#L122) is also wrong (the method name explicitly mentions "rotation" and not "quaternion angle"), and is missing the factor 2.

(Doesn't change anything for our case since the slerp operation explicitly requires the angle between quaternions, not rotations)